### PR TITLE
MDCT-1870: Drawer Component

### PR DESCRIPTION
**Pull request by karla-vm**
## Description

Creating a Drawer component based off of ChakraUI's [Drawer](https://chakra-ui.com/docs/components/drawer) documentation and the existing MCR Modal component.

Paired on this with @juliuscaesar ! 💻 

### How to test

N/A

### Changed Dependencies

N/A

## Code author checklist
- [ ] I have performed a self-review of my code
- [ ] I have added [thorough](https://bit.ly/3zPrxuZ) tests
- [ ] I have added analytics, if necessary
- [ ] I have updated the documentation, if necessary

## Reviewer checklist (two different people)
- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

### CODE REVIEW

File: services/ui-src/src/components/drawer/Drawer.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+          {/* TODO: Get Form details somehow */}
+          {/* <Form /> */}
+          <form>Placeholder</form>
+        </DrawerBody>
+        <DrawerFooter sx={sx.drawerFooter}>
+          <Button sx={sx.action}>Save</Button>
+          <Button sx={sx.close} onClick={drawerState.onClose}>
+            Close
+          </Button>
+        </DrawerFooter>
```
Comment (ntsummers1) [inline services/ui-src/src/components/drawer/Drawer.tsx:56]: I've updated the modal component on the dashboard to do the following:
```
        <ModalBody>{children}</ModalBody>

        <ModalFooter>
          <Button
            onClick={!actionId ? () => actionFunction() : undefined}
            form={actionId}
            type="submit"
          >
            {content.actionButtonText}
          </Button>
          <Button
            variant="link"
            onClick={modalState.onClose}
          >
            {content.closeButtonText}
          </Button>
        </ModalFooter>
```

That lets you pass in a form as a child in the parent component like so:

```
      <Modal>
        <Form
          id={formJson.id}
          formJson={formJson}
          formSchema={formSchema}
          onSubmit={addProgram}
        />
      </Modal>
```

Now that code will work though it hasn't been reviewed. Might be helpful though?

Comment (karla-vm) [reply services/ui-src/src/components/drawer/Drawer.tsx:56]: > 

That worked! Thanks, @ntsummers1 🥇 

Screen Shot 2022-08-17 at 9 32 43 AM

**Comment by braxex** [comment]
@juliuscaesar @karla-vm 

Wrote some tests for what we've done so far. This is getting so big that I think we ought to call it for now and merge this, then continue in another branch. 

In addition to writing tests for everything, I fragmented McparReportPage out.

Let me know what you think. 👍

**Comment by braxex** [comment]
Note: The failing CodeClimate check is not a big deal in this instance. Please disregard it and we'll override it when it's time to merge.

**Review by braxex — APPROVED** [review]

**Pull request merged**